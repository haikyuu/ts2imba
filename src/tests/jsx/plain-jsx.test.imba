import {build} from "../../index"
import * as imbac from 'imba/compiler'

let tsx-code = '''
function App(){
	return (
		<div>
			<span>Test</span>
		</div>
	)
}
'''

let imba-code = '''
tag App
	def render
		<self>
			<div>
				<span>
					"Test"

'''
test 'spaces_4' do
	const result = await build tsx-code
	expect(result.code).toEqual(imba-code)
	const out = imbac.compile(result.code, sourceId: 'sth')