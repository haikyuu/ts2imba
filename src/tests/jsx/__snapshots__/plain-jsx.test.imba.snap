// Vitest Snapshot v1

exports[`spaces_4 1`] = `
"tag App
	def render
		<self>
			<div>
				<span>
					\\"Test\\"
"
`;
